### PR TITLE
Support _NET_WM_STATE_FOCUSED

### LINKED ISSUE #2273: Support _NET_WM_STATE_FOCUSED
Output of `i3 --moreversion 2>&- || i3 --version`:

i3 version 4.12 (2016-03-06, branch "4.12") © 2009 Michael Stapelberg and contributors

**What I did:**

Run any gtk3 app with client side decoration like Nautilus or Evince. Make window floating, Move focus to this window and move out.

**What I saw:**

Window decoration stays like window is focused.

**What I expected instead:**

Window decoration should change appearance depending of is window focused.

Xfwm4 patch https://bugzilla.xfce.org/show_bug.cgi?id=11067
Gnome bug https://bugzilla.gnome.org/show_bug.cgi?id=734159

### PR DESCRIPTION
**Pull request by orestisf1993**
Continuation of #2314.

Fixes #2273.

I admit that I don't fully understand all the gory details but the original PR seemed to be close to being accepted and I'd like to see this merged with i3. The fixup commit has my changes from the original discussion.

### CODE REVIEW

File: testcases/t/264-net-wm-state-focused.t [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+$windowB = open_window;
+
+ok(!is_focused($windowA), 'when a another window is focused, the old window should not have _NET_WM_STATE_FOCUSED set');
+
+fresh_workspace;
+
+ok(!is_focused($windowB), 'when focus moves to the ewmh support window, neither window should have _NET_WM_STATE_FOCUSED set');
+
+# TODO: test to make sure the atom is properly set/unset when WM_TAKE_FOCUS is used to focus the window
```
Comment (orestisf1993) [inline testcases/t/264-net-wm-state-focused.t]: `WM_TAKE_FOCUS` is tested in `158-wm_take_focus.t`. Should I add a test there?

Comment (Airblader) [reply testcases/t/264-net-wm-state-focused.t]: Either place is fine with me.

Comment (orestisf1993) [reply testcases/t/264-net-wm-state-focused.t]: A problem is that both `WM_TAKE_FOCUS` tests and `_NET_WM_STATE_FOCUSED` tests rely on a big subroutine. Any way I can make it less ugly?

Comment (Airblader) [reply testcases/t/264-net-wm-state-focused.t]: We can move it to the i3test library

Comment (orestisf1993) [reply testcases/t/264-net-wm-state-focused.t]: Ok. We could also create a subroutine `is_focused` that checks with `get_focused` and `_NET_WM_STATE_FOCUSED` and use this globally. Might be overkill though.

**Comment by orestisf1993** [comment]
up

File: src/x.c [hunk 2]
```diff
@@ -99,6 +99,21 @@ static con_state *state_for_frame(xcb_window_t window) {
...
+/*
+ * Changes the atoms on the root window and the windows themselves to properly
+ * reflect the current focus for ewmh compliance.
+ *
+ */
+static void change_ewmh_focus(xcb_window_t new_focus, xcb_window_t old_focus) {
+    ewmh_update_active_window(new_focus);
+
+    if (new_focus != XCB_WINDOW_NONE)
+        ewmh_update_focused(focused->window->id, true);
```
Comment (Airblader) [inline src/x.c]: Should this not be using `new_focus`?

Comment (orestisf1993) [reply src/x.c]: Yes, up

**Pull request merged**